src/main/webapp/app/pages/companyPage/CompanyPage.tsx

```diff
@@ -0,0 +1,521 @@
+
+interface MatchParams {
+  id: string;
+}
+
+interface ICompanyPageProps extends RouteComponentProps<MatchParams> {
+  windowStore: WindowStore;
+}
+
+type SelectOptionType = {
+  label: string;
+  value: number;
+  user: UserDTO;
+};
+
+const LICENSE_STATUS_UPDATE_MESSAGES = {
+  REGULAR: {
+    TRIAL: `All users will be sent a trial activation email.`,
+    EXPIRED: 'All accounts will be terminated immediately.',
+  },
+  TRIAL: {
+    REGULAR: 'All accounts will be converted to a regular license.',
+    TRIAL_EXPIRED: 'All trial accounts will be terminated immediately.',
+    EXPIRED: 'All trial accounts will be terminated immediately',
+  },
+  TRIAL_EXPIRED: {
+    TRIAL: 'All accounts can be reactivated using the activation email.',
+    REGULAR: 'All account will be converted to a regular license.',
+  },
+  EXPIRED: {
+    TRIAL: 'All account will be sent a trial activation email.',
+    REGULAR: 'All accounts will be approved automatically.',
+  },
+};
+
+@observer
+export default class CompanyPage extends React.Component<ICompanyPageProps> {
+  @observable company: CompanyDTO;
+  @observable getCompanyStatus: PromiseStatus;
+  @observable selectedLicenseStatus: LicenseStatus;
+
+  @observable showModal = false;
+  @observable confirmModalText = '';
+  @observable formValues: any;
+
+  @observable companyUsers: UserDTO[] = [];
+  @observable companyUsersTokens: Array<Token[]> = [];
+  @observable availableUsers: SelectOptionType[] = [];
+  @observable selectedUsersOptions: SelectOptionType[] = [];
+
+  constructor(props: ICompanyPageProps) {
+    super(props);
+    this.getCompanyInfos();
+  }
+
+  @action
+  async getCompanyInfos() {
+    try {
+      this.getCompanyStatus = PromiseStatus.pending;
+      this.company = await client.getCompanyUsingGET({
+        id: parseInt(this.props.match.params.id),
+      });
+      this.selectedLicenseStatus = this.company.licenseStatus as LicenseStatus;
+      await this.getCompanyUsers();
+      await this.getDropdownUsers();
+      await this.getAllUsersTokens();
+      this.getCompanyStatus = PromiseStatus.complete;
+    } catch (error) {
+      this.getCompanyStatus = PromiseStatus.error;
+      notifyError(error);
+    }
+  }
+
+  @action
+  async getCompanyUsers() {
+    try {
+      this.companyUsers = await client.getCompanyUsersUsingGET({
+        id: this.company.id,
+      });
+    } catch (error) {
+      notifyError(error);
+    }
+  }
+
+  @action
+  async getDropdownUsers() {
+    const allUsers = await client.getAllUsersUsingGET({
+      size: THRESHOLD_NUM_OF_USER,
+    });
+    this.availableUsers = allUsers
+      .filter(
+        user =>
+          !this.companyUsers.some(companyUser => companyUser.id === user.id)
+      )
+      .map(user => ({
+        label: user.email,
+        value: user.id,
+        user,
+      }));
+  }
+
+  @action
+  async getAllUsersTokens() {
+    for (const user of this.companyUsers) {
+      const tokens = await client.getUserTokensUsingGET({
```

Comment:
This is ok for now, but might below the performance when too many users are associated with the company. Let's leave it as it but could you add a comment here?

Reply:
From weekly meeting: Create another endpoint that will take a list of users and return a list of users tokens